Print all styles

msabramo opened the pull request:
1. Print available styles in several groups, one per line (textStyles, colorStyles, brightColorStyles, bgColorStyles, and bgBrightColorStyles).

2. Show available styles when invalid style provided (with a bit of extra color too for the error message).

![Screen Shot 2021-09-22 at 4 02 18 AM](https://user-images.githubusercontent.com/305268/134332434-b7716e10-c4d2-4bbd-8104-1c03b6f209b5.png)

![Screen Shot 2021-09-22 at 4 01 55 AM](https://user-images.githubusercontent.com/305268/134332480-59a1fa75-1b70-46b0-b4a1-386309c3095d.png)

msabramo committed:
Print available styles in 3 groups

One per line:

1. textStyles
2. colorStyles
3. bgColorStyles

Break style on to 3 lines, rather than one very long line.

msabramo committed:
Show available styles when invalid style provided

It may help the user to know what the valid styles are.

sindresorhus commented on cli.js:
```diff
@@ -32,8 +32,22 @@ const printAllStyles = () => {
 		'bgCyan',
 		'bgWhite',
 	];
+	allStyles.sliceByValue = function (startValue, endValue) {
```
It's better to accept an array as argument than to add a property to the array.

msabramo replied:
Just updated and the code no longer uses `sliceByValue` at all.

sindresorhus commented:
Have a look at https://github.com/chalk/chalk/blob/main/examples/screenshot.js, which does the same thing.

msabramo committed:
Make sliceByValue accept an array as argument

rather than adding a property to the array.

msabramo committed:
Use ansiStyles & include bright colors

msabramo committed:
Borrow blackifying code for chalk screenshot

Add code to blackify the text for most of the `bg` styles, inspired by
https://github.com/chalk/chalk/blob/main/examples/screenshot.js

msabramo committed:
Go back to hard-coding styles

instead of using them from external `ansi-styles` package. The styles
and ordering of the styles in that package can change too easily and
break our code here, which relies on a specific ordering.

msabramo commented:
Ready for review.

sindresorhus commented:
Some of the foreground texts are unreadable. This is not the case for the Chalk repo demo.

msabramo commented:
> Some of the foreground texts are unreadable. This is not the case for the Chalk repo demo.

Which ones in particular?

msabramo commented:
I think how good this looks and which ones are hard to see will be highly dependent on the user’s terminal color settings. 

Did you think some of the ones in my screenshot were unreadable or are they unreadable on your terminal and if so I guess a screenshot could be helpful.

sindresorhus commented:
Never mind. Git had screwed up and had preserved the changes from your last commits even when I updated. The latest changes look good to me.

msabramo commented:
Ah okay. Well I was going to post this screenshot I just took that compares the output of `chalk --demo` with this PR to the chalk screenshot script. I guess it's not quite as important now, but since I already took it and it might be interesting to look at, here it is:

![Screen Shot 2021-09-23 at 10 17 20 AM](https://user-images.githubusercontent.com/305268/134554134-bca20163-6dce-4794-99ab-14758b63dc8c.png)
